includes/exceptions.h

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#if defined(OCOS_NO_EXCEPTIONS) || defined(OCOS_CONTAIN_EXCEPTIONS)
+#if defined(__ANDROID__)
+#include <android/log.h>
+#else
+#include <iostream>
+#endif
+#endif
+
+#include <stdexcept>
+
+#include "onnxruntime_c_api.h"
+
+namespace OrtW {
+
+// All C++ methods that can fail will throw an exception of this type
+struct Exception : std::exception {
+  Exception(std::string message, OrtErrorCode code) : message_{std::move(message)}, code_{code} {}
+
+  OrtErrorCode GetOrtErrorCode() const { return code_; }
+  const char* what() const noexcept override { return message_.c_str(); }
+
+ private:
+  std::string message_;
+  OrtErrorCode code_;
+};
+
+#if defined(OCOS_NO_EXCEPTIONS) || defined(OCOS_CONTAIN_EXCEPTIONS)
+inline void PrintFinalMessage(const char* file, int line, const char* msg) {
+#if defined(__ANDROID__)
+  __android_log_print(ANDROID_LOG_ERROR, "onnxruntime-extensions", "Exception in %s line %d: %s", file, line, msg);
+#else
+  std::cerr << "Exception in " << file << " line " << line << ": " << msg << std::endl;
```

Comment:
Would fprintf(stderr, ....) be safer?

Reply:
Typically in C++ using IO streams  is safer and the recommended approach as all the arguments are type checked at compile type.

https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#slio3-prefer-iostreams-for-io

There's no performance reason here to use printf.

Reply:
@wenbingl Please let me know how I can re-create the warning from using std::cerr. Can change to printf if we can't fix that warning.

Reply:
Some warning like this,
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.35.32215\include\ostream(360,1): warning C4530: C++ exception handler used, but unwind semantics are not enabled. Specify /EHsc

But, I just figure out defining a _HAS_EXCEPTIONS=0 can remove the warnings from MSVC, but not sure if it is officially supported and how about other compilers,